src/lib.rs

```diff
@@ -20,49 +52,75 @@ impl Cleverbot {
     /// Creates a new Cleverbot instance.
     /// * `user` - The API User.
     /// * `key` - The API Key.
-    pub fn new(user: String, key: String, nick: Option<String>) -> Result<Cleverbot, String> {
+    pub fn new(user: String, key: String, nick: Option<String>) -> Result<Cleverbot, CleverbotError> {
         let mut response = {
             let mut args = vec![
                 ("user", &*user),
                 ("key", &*key),
             ];
-            if let Some(ref nick) = nick { args.push(("nick", &*nick)) }
-            request("https://cleverbot.io/1.0/create", &*args)
+            if let Some(ref nick) = nick { args.push(("nick", &*nick)) };
+            try!(request("https://cleverbot.io/1.0/create", &*args))
         };
         let mut body = String::new();
-        response.read_to_string(&mut body).unwrap();
-        let json: BTreeMap<String, String> = serde_json::from_str(&body).unwrap();
-        let result = json.get("status").unwrap();
-        if result == "success" {
-            Ok(Cleverbot {
-                user: user,
-                key: key,
-                nick: json.get("nick").unwrap().to_string(),
-            })
-        } else {
-            Err(result.to_string())
+        try!(response.read_to_string(&mut body));
+        let json: BTreeMap<String, String> = try!(serde_json::from_str(&body));
+        let opt_result = json.get("status");
+        let result = match opt_result {
+            Some(result) => result,
+            None => return Err(CleverbotError::MissingValue(String::from("status"))),
+        };
+        match result.as_ref() {
+            "success" => {
+                let json_nick = json.get("nick");
+                let nick = match json_nick {
+                    Some(nick) => nick,
```

Comment:
You could make this the `Ok(Cleverbot` bit, letting you make that `return Err` below an implicit return.